crates/relayer-cli/src/commands/listen.rs

```diff
@@ -144,29 +144,25 @@ fn subscribe(
     compat_mode: CompatMode,
     rt: Arc<TokioRuntime>,
 ) -> eyre::Result<Subscription> {
-    let EventSourceMode::Push { url, batch_delay } = &chain_config.event_source else {
-        return Err(eyre!("unsupported event source mode, only 'push' is supported for listening to events"));
-    };
-
-    let (mut event_source, tx_cmd) = EventSource::new(
-        chain_config.id.clone(),
-        url.clone(),
-        compat_mode,
-        *batch_delay,
-        rt,
-    )
-    .map_err(|e| eyre!("could not initialize event source: {}", e))?;
-
-    event_source
-        .init_subscriptions()
-        .map_err(|e| eyre!("could not initialize subscriptions: {}", e))?;
-
-    let queries = event_source.queries();
-    info!("listening for queries: {}", queries.iter().format(", "),);
-
-    thread::spawn(|| event_source.run());
-
-    let subscription = tx_cmd.subscribe()?;
+    let (event_source, monitor_tx) = match &chain_config.event_source {
+        EventSourceMode::Push { url, batch_delay } => EventSource::websocket(
+            chain_config.id.clone(),
+            url.clone(),
+            compat_mode,
+            *batch_delay,
+            rt.clone(),
```

Comment:
```suggestion
            rt,
```
This seems like a redundant clone